backend/src/v5/processors/teamspaces/projects/models/commons/modelList.js

```diff
@@ -15,12 +15,59 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+const { v4Path } = require('../../../../../../interop');
+// eslint-disable-next-line require-sort/require-sort
+const {
+	addModel,
+	deleteModel,
+	getModelById,
+	getModelByName,
+	isSubModel,
+	removeModelCollections,
+} = require('../../../../../models/modelSettings');
+const { addProjectModel, getProjectById, removeProjectModel } = require('../../../../../models/projects');
 const { hasProjectAdminPermissions, isTeamspaceAdmin } = require('../../../../../utils/permissions/permissions');
 const { getFavourites } = require('../../../../../models/users');
-const { getProjectById } = require('../../../../../models/projects');
+const { logger } = require('../../../../../utils/logger');
+// eslint-disable-next-line import/no-dynamic-require, security/detect-non-literal-require
+const { removeAllFilesFromModel } = require(`${v4Path}/models/fileRef`);
+const { templates } = require('../../../../../utils/responseCodes');
 
 const ModelList = {};
 
+ModelList.addModel = async (teamspace, project, user, data) => {
+	const { models } = await getProjectById(teamspace, project, { models: 1 });
+	const modelSetting = await getModelByName(teamspace, models, data.name, { _id: 0, name: 1 });
+
+	if (modelSetting) {
+		throw templates.nameAlreadyExists;
+	}
+
+	const insertedId = addModel(teamspace, data);
+
+	await addProjectModel(teamspace, project, insertedId);
+
+	return insertedId;
+};
+
+ModelList.deleteModel = async (teamspace, project, model) => {
+	const setting = await getModelById(teamspace, model);
+
+	if (!setting.federate && (await isSubModel(teamspace, model))) {
+		throw templates.containerIsSubModel;
+	}
+
+	try {
+		await removeAllFilesFromModel(teamspace, model);
+	} catch (err) {
+		logger.logDebug(`[Delete model] : ${JSON.stringify(err)}`);
+	}
+
+	await removeModelCollections(teamspace, model);
```

Comment:
I don't think there's any dependencies between these 3, so they should just go into a Promise.all() so we don't serialise it.